protocol/parser/parser.go

```diff
@@ -510,21 +530,28 @@ func blockInterfaceToString(block interface{}) string {
 	}
 }
 
-func parseByArg(rpcInput RPCInput, input []string, dataSource int) ([]interface{}, error) {
+func parseByArg(rpcInput RPCInput, input []string, dataSource int, parserOptions *ParserOptions) ([]interface{}, error) {
+	var lavaLogSeverity uint = utils.LAVA_LOG_PRODUCTION
+	if !parserOptions.shouldWarnLogOnFailedParsing() {
+		lavaLogSeverity = utils.LAVA_LOG_DEBUG
+	}
+
 	// specified block is one of the direct parameters, input should be one string defining the location of the block
 	if len(input) != 1 {
-		return nil, utils.LavaFormatProduction("invalid input format, input length", nil, utils.LogAttr("input_len", strconv.Itoa(len(input))))
+		utils.LavaFormatLog("invalid input format, input length", nil, lavaslices.Slice(utils.LogAttr("input_len", len(input))), lavaLogSeverity)
```

Comment:
I rather instead of sending logger stuff to all functions. just return fmt.Error without printing in all parser functions. and decide wether we need to print or not outside. apply it to all changes.

Reply:
Fixed in effb633f002038df0c5d191361cd2205e1d6ca35